tests/pass/tree_borrows/cell-inside-struct.rs

```diff
@@ -0,0 +1,34 @@
+//! The same as `tests/fail/tree-borrows/cell-inside-struct` but with
+//! precise tracking of interior mutability disabled.
+//@compile-flags: -Zmiri-tree-borrows -Zmiri-tree-borrows-no-precise-interior-mut
```

Comment:
I think this still needs to disable the GC?

Reply:
Now the tree printed is also [compacted](https://github.com/rust-lang/miri/pull/4376/files/1aeac15008767ba13a8ea157f1811a4318054c37..adf6838d1aa39407004229fb2c138ddf0ebbdabf#diff-da183195935df371d899392cf40694d044d25cd22986fa1bd4ae5bbf6ab9f028), so I don't think this is necessary?

Reply:
Yes then it presumably works.